src/include/str_list.h

```diff
@@ -117,9 +117,9 @@ inline std::string str_join(const std::vector<std::string>& v, const std::string
 {
   if (v.empty())
     return std::string();
-  std::vector<std::string>::const_iterator i = v.begin();
+  auto i = v.cbegin();
```

Comment:
please note, standard library offers:
```c++
const_iterator begin() const;
```
and 

```c++
const_iterator begin() const noexcept;
```

there are lots of places we can replace a type with `auto`. i'd avoid such kind of change unless you happen to work on it.

Reply:
@tchaikov 
The purpose here is not to use "auto". I want to change it to use ```cbegin()``` instead of ```begin()```
BTW, I don't know whether it's proper for me to file such kind of PR. I just want to make it easy for me to understand it.

Reply:
the commit message reads

> common/str_list: simplify define vector const iterator 

I cannot see why cbegin is simpler than begin. Or if helps with the readability.